src/Microsoft.IdentityModel.Tokens/AsymmetricSecurityKey.cs

```diff
@@ -29,8 +29,7 @@ namespace Microsoft.IdentityModel.Tokens
 {
     public abstract class AsymmetricSecurityKey : SecurityKey
     {
-        public abstract bool HasPrivateKey { get; }
+        public abstract bool? HasPrivateKey { get; }
```

Comment:
This is a mistake. I wanted the public API to return bool only and have the internal implementation use bool?. I will correct this. Thanks for pointing out.